src/utils/api.ts

```diff
@@ -1,7 +1,24 @@
+import { tryGetString } from './types'
+
 interface APIError {
   error: { detail: string }
 }
 
 export const isHTTPError = (e: unknown): e is APIError => {
   return (e as APIError).error !== undefined
 }
+
+export const getHumanReadableError = (e: unknown, defaultErrorMessage: string) => {
+  const stringifiedError = tryGetString(e)
+  let returnedError
+
+  if (isHTTPError(e)) {
+    returnedError = e.error.detail
+  } else if (stringifiedError) {
+    returnedError = stringifiedError
+  } else {
+    returnedError = defaultErrorMessage
+  }
+
+  return returnedError
```

Comment:
We could simplify the code to this:

```js
export const getHumanReadableError = (e: unknown, defaultErrorMessage: string) => {
  const stringifiedError = tryGetString(e)

  if (isHTTPError(e)) {
    return e.error.detail
  } else if (stringifiedError) {
    return stringifiedError
  }

  return defaultErrorMessage
}
```

or if you feel fancy, this:

```js
export const getHumanReadableError = (e: unknown, defaultErrorMessage: string) => {
  const stringifiedError = tryGetString(e)
  
  return isHTTPError(e) ? e.error.detail : stringifiedError || defaultErrorMessage
}
```

No strong preference, I leave it up to you :)

Reply:
Let's go with the ternary version (I hesitated, went for the explicit version, but let's change my mind ;)). I'll make the change.